src/data_structures/suffix_array.rs

```diff
@@ -24,6 +23,7 @@ use crate::data_structures::smallints::SmallInts;
 
 pub type LCPArray = SmallInts<i8, isize>;
 pub type RawSuffixArray = Vec<usize>;
+pub type RawSuffixArraySlice<'a> = &'a [usize];
```

Comment:
This (and the downstream changes) also seems like a bigger change. Was this simply a clippy suggestion?

Reply:
It's just a type alias, doesn't really change anything, apart from making it possible to pass any collection that can be viewed as a slice of usize